File: app/src/main/java/com/example/watering/investrecord/DBHelper/DBHelper.java
package com.example.watering.investrecord.DBHelper;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteException;
import android.database.sqlite.SQLiteOpenHelper;
import android.provider.BaseColumns;

/**
 * Created by watering on 17. 10. 21.
 */

@SuppressWarnings("DefaultFileTemplate")
public class DBHelper extends SQLiteOpenHelper {

    private static final int db_version = 7;
    private static final String DB_FILE_NAME = "InvestRecord.db";
    String TABLE_NAME;
    String [] COLUMNS;

    DBHelper(Context context) {
        super(context, DB_FILE_NAME, null, db_version);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        StringBuilder sql;

        sql = new StringBuilder("CREATE TABLE IF NOT EXISTS " + TABLE_NAME + " ("
                + BaseColumns._ID + " INTEGER PRIMARY KEY AUTOINCREMENT ");

        for (String COLUMN : COLUMNS) {
            sql.append(", ").append(COLUMN);
        }
        sql.append(");");
        db.execSQL(sql.toString());
    }
    @Override
    public void onOpen(SQLiteDatabase db) {
        super.onOpen(db);

        StringBuilder sql;

        sql = new StringBuilder("CREATE TABLE IF NOT EXISTS ").append(TABLE_NAME).append(" (")
                .append(BaseColumns._ID).append(" INTEGER PRIMARY KEY AUTOINCREMENT");

        for (String COLUMN : COLUMNS) {
            sql.append(", ").append(COLUMN);
        }
        sql.append(");");
        db.execSQL(sql.toString());
    }
    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL("DROP TABLE IF EXISTS " + TABLE_NAME);
        onCreate(db);
    }

    public void insert(ContentValues values) throws SQLiteException {
        getWritableDatabase().insert(TABLE_NAME, null, values);
    }
    public void delete(String where, String[] whereArgs) throws SQLiteException {
        String selection;

        if(where == null) {
            selection = null;
        }
        else {
            selection = where + "=?";
        }
        getWritableDatabase().delete(TABLE_NAME, selection, whereArgs);
    }
    public void update(ContentValues values, String where, String[] selectionArgs) throws SQLiteException {
        getWritableDatabase().update(TABLE_NAME, values, where + "=?", selectionArgs);
    }
    public Cursor query(String[] columns, String selection, String[] selectionArgs, String orderBy) throws SQLiteException {
        return getReadableDatabase().query(TABLE_NAME, columns, selection, selectionArgs, null, null, orderBy);
    }
}

File: app/src/main/java/com/example/watering/investrecord/DBHelper/DBHelper.kt
package com.example.watering.investrecord.DBHelper

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteException
import android.database.sqlite.SQLiteOpenHelper
import android.provider.BaseColumns

/**
 * Created by watering on 17. 10. 21.
 */

open class DBHelper internal constructor(context: Context) : SQLiteOpenHelper(context, DB_FILE_NAME, null, db_version) {
    internal var TABLE_NAME: String? = null
    internal var COLUMNS: Array<String>? = null

    override fun onCreate(db: SQLiteDatabase) {
        val sql: StringBuilder

        sql = StringBuilder("CREATE TABLE IF NOT EXISTS " + TABLE_NAME + " ("
                + BaseColumns._ID + " INTEGER PRIMARY KEY AUTOINCREMENT ")

        for (COLUMN in COLUMNS!!) {
            sql.append(", ").append(COLUMN)
        }
        sql.append(");")
        db.execSQL(sql.toString())
    }

    override fun onOpen(db: SQLiteDatabase) {
        super.onOpen(db)

        val sql: StringBuilder

        sql = StringBuilder("CREATE TABLE IF NOT EXISTS ").append(TABLE_NAME).append(" (")
                .append(BaseColumns._ID).append(" INTEGER PRIMARY KEY AUTOINCREMENT")

        for (COLUMN in COLUMNS!!) {
            sql.append(", ").append(COLUMN)
        }
        sql.append(");")
        db.execSQL(sql.toString())
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL("DROP TABLE IF EXISTS " + TABLE_NAME!!)
        onCreate(db)
    }

    @Throws(SQLiteException::class)
    fun insert(values: ContentValues) {
        writableDatabase.insert(TABLE_NAME, null, values)
    }

    @Throws(SQLiteException::class)
    fun delete(where: String?, whereArgs: Array<String>) {
        val selection: String?

        if (where == null) {
            selection = null
        } else {
            selection = "$where=?"
        }
        writableDatabase.delete(TABLE_NAME, selection, whereArgs)
    }

    @Throws(SQLiteException::class)
    fun update(values: ContentValues, where: String, selectionArgs: Array<String>) {
        writableDatabase.update(TABLE_NAME, values, "$where=?", selectionArgs)
    }

    @Throws(SQLiteException::class)
    fun query(columns: Array<String>?, selection: String?, selectionArgs: Array<String>?, orderBy: String?): Cursor {
        return readableDatabase.query(TABLE_NAME, columns, selection, selectionArgs, null, null, orderBy)
    }

    companion object {

        private val db_version = 7
        private val DB_FILE_NAME = "InvestRecord.db"
    }
}
